control-plane-operator/controllers/hostedcontrolplane/oapi/deployment_test.go

```diff
@@ -0,0 +1,142 @@
+			Data: map[string]string{"config.yaml": "test-data"},
+		}
+		volumeProjection = []corev1.VolumeProjection{}
+	)
+	hcp.Name = "name"
+	hcp.Namespace = "namespace"
+	ownerRef := config.OwnerRefFrom(hcp)
+	testCases := []struct {
+		name                         string
+		cm                           corev1.ConfigMap
+		expectedVolume               *corev1.Volume
+		auditConfig                  *corev1.ConfigMap
+		expectedVolumeProjection     []corev1.VolumeProjection
+		deploymentConfig             config.DeploymentConfig
+		additionalTrustBundle        *corev1.LocalObjectReference
+		clusterConf                  *hyperv1.ClusterConfiguration
+		expectProjectedVolumeMounted bool
+	}{
+		{
+			name:             "Trust bundle provided",
+			auditConfig:      manifests.OpenShiftAPIServerAuditConfig(targetNamespace),
+			deploymentConfig: config.DeploymentConfig{},
+			additionalTrustBundle: &corev1.LocalObjectReference{
+				Name: "user-ca-bundle",
+			},
+			expectedVolume: &corev1.Volume{
+				Name: "additional-trust-bundle",
+				VolumeSource: corev1.VolumeSource{
+					Projected: &corev1.ProjectedVolumeSource{
+						Sources: append(volumeProjection, getFakeVolumeProjectionCABundle()),
+					},
+				},
+			},
+			clusterConf:                  nil,
+			expectProjectedVolumeMounted: true,
+		},
+		{
+			name:                         "Trust bundle not provided",
+			auditConfig:                  manifests.OpenShiftAPIServerAuditConfig(targetNamespace),
+			deploymentConfig:             config.DeploymentConfig{},
+			expectedVolume:               nil,
+			additionalTrustBundle:        nil,
+			clusterConf:                  nil,
+			expectProjectedVolumeMounted: false,
+		},
+		{
+			name:             "Trust bundle and clusterConf provided",
+			auditConfig:      manifests.OpenShiftAPIServerAuditConfig(targetNamespace),
+			deploymentConfig: config.DeploymentConfig{},
+			additionalTrustBundle: &corev1.LocalObjectReference{
+				Name: "user-ca-bundle",
+			},
+			expectedVolume: &corev1.Volume{
+				Name: "additional-trust-bundle",
+				VolumeSource: corev1.VolumeSource{
+					Projected: &corev1.ProjectedVolumeSource{
+						Sources: append(volumeProjection, getFakeVolumeProjectionCABundle(), getFakeVolumeProjectionClusterConf()),
+					},
+				},
+			},
+			clusterConf: &hyperv1.ClusterConfiguration{
+				Image: &configv1.ImageSpec{
+					AdditionalTrustedCA: configv1.ConfigMapNameReference{
+						Name: "sample-cluster-conf-image-ca",
+					},
+				},
+			},
+			expectProjectedVolumeMounted: true,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			g := NewGomegaWithT(t)
+			tc.auditConfig.Data = map[string]string{"policy.yaml": "test-data"}
+			err := ReconcileDeployment(oapiDeployment, nil, ownerRef, testOapiCM, tc.auditConfig, nil, tc.deploymentConfig, imageName, "socks5ProxyImage", config.DefaultEtcdURL, util.AvailabilityProberImageName, false, hyperv1.AgentPlatform, tc.additionalTrustBundle, tc.clusterConf)
```

Comment:
this is also sharing  oapiDeployment across test cases.
Can we please follow up to clean it up on main and understand why the other unit was failing?